doc/roadmap.md

```diff
@@ -0,0 +1,92 @@
+# Roadmap
+
+This page describes a tentative roadmap for hvPlot in 2024
+
+## 1. Documentation, documentation, documentation
+
+### Context
+
+hvPlot's website currently has the following structure:
+
+- A landing page describing its main features.
+- A *Getting Started* guide (recently added) that is a short tutorial demonstrating its basic capabilities.
+- Many *User Guides*, focusing on a domain (e.g. *Geographic Data*), on a data type (*Gridded Data*), on a feature (e.g. *Interactive*), or on a specific part of the API (e.g. *Customization*).
+- A *Reference Gallery* with a page per plot method (actually incomplete), separated by these categories: Pandas, GeoPandas, and Xarray. Each reference page is usually short, only uses Bokeh, and doesn't include the whole API applicable to the plot method.
+- A *Topics* short page containing a few links to other websites using hvPlot
+- A *Developer Guide*, a *Releases* page and an *About* page
+
+hvPlot users have repeatedly complained about the quality of its documentation, which seems to mostly stem from:
+
+- The API is not being documented.
+- The API is not being enough demonstrated in a single place; users have to assemble examples from different pages.
+- An implicit reliance on HoloViews' documentation (e.g. for the options) that itself isn't in great shape.
+- The difficulty to run locally all or parts of the documentation (that relies on somewhat complex datasets and many libraries).
```

Comment:
Maybe the discovery of available kwargs/options could be improved by using an LLM for hvplot like:
https://huggingface.co/spaces/ahuang11/tweak-mpl-chat

Reply:
Yes sure LLMs could help users in some ways! How exactly I'm not sure.

I've tried to keep the roadmap simple, it focuses on the boring but needed stuff I believe we have to address.

Reply:
I made a basic LLM chat bot for hvplot here: https://huggingface.co/spaces/ahuang11/tweak-hvplot-chat

it's missing a lot of docs because it was a pain to convert ipynb to markdown, but it has the basics of customization https://huggingface.co/spaces/ahuang11/tweak-hvplot-chat/tree/main/hvplot_docs

Reply:
Hi @ahuang11 

Did you consider using and improving the existing hvplot chat app? https://holoviz-topics.github.io/panel-chat-examples/openai/#chat-with-hvplot

Reply:
If I'm not mistaken I believe that one is limited by formatting the kwargs into JSON so openai can do function calling. The one I just implemented is to dump hvPlot/HoloViews docs into a directory, so it expands a bit into HoloViews and can use .opts (it could probably write hooks as well if I gave it the FAQ).